src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -656,6 +656,20 @@ private async Task<bool> EnumerateFilesFromArtifactsProvider(TContext globalCont
 
                 string filePath = artifact.Uri.GetFilePath();
 
+                long artifactSizeInBytes = -1;
```

Comment:
NIT: Should this block be moved a little down just before the statement 669, where the resulting artifactSizeInBytes will be evaluated (whether it is 0 or whether it is exceeding the size threshold?

Reply:
I hope that placing it up at the top of the method like this sends the message to future maintainers that they should tend to add new functionality _below_ it.  This reduces the risk of a file access that would trigger one of the exceptions winding up before the check that's guarded by the catch blocks.